src/Data/StringVariants/NonEmptyText.hs

```diff
@@ -82,15 +83,16 @@ nonEmptyTextToText (NonEmptyText t) = t
 -- | Identical to the normal text filter function, but maintains the type-level invariant
 -- that the text length is <= n, unlike unwrapping the text, filtering, then
 -- rewrapping the text.
-filterNonEmptyText :: (Char -> Bool) -> NonEmptyText n -> NonEmptyText n
-filterNonEmptyText f (NonEmptyText t) = NonEmptyText (T.filter f t)
+filterNonEmptyText :: KnownNat n => (Char -> Bool) -> NonEmptyText n -> Maybe (NonEmptyText n)
+filterNonEmptyText f (NonEmptyText t) = mkNonEmptyText (T.filter f t)
 
 -- | Narrows the maximum length, dropping any remaining trailing characters.
-takeNonEmptyText :: forall m n. (KnownNat m, KnownNat n, n <= m) => NonEmptyText m -> NonEmptyText n
+takeNonEmptyText :: forall m n. (KnownNat m, KnownNat n, n <= m, 0 <= n) => NonEmptyText m -> NonEmptyText n
```

Comment:
This would permit `NonEmptyText 0`, which seems wrong to me?

```suggestion
takeNonEmptyText :: forall m n. (KnownNat m, KnownNat n, n <= m, 0 < n) => NonEmptyText m -> NonEmptyText n
```

Reply:
Yep, that change was meant to fix it.